### PR TITLE
intstr: add constructor from int32, deprecate int constructor

### PR DESCRIPTION
**Pull request by skitt**
#### What type of PR is this?

/kind cleanup
/kind api-change

#### What this PR does / why we need it:

`intstr` wraps an `int32` but exposes an `int`-based API. This PR adds a `FromInt32` constructor, deprecates the `FromInt` one, migrates simple uses to the new constructor.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/ok-to-test

but probably should merge after code-freeze

**Comment by skitt** [comment]
> /ok-to-test
> 
> but probably should merge after code-freeze

Definitely, I pushed this because it was ready but it should wait — I’ll rebase as necessary.

**Comment by skitt** [comment]
@alculquicondor it’s after code-freeze, do you know who should review this?

**Comment by alculquicondor** [comment]
could you split this by SIG?

**Comment by skitt** [comment]
> could you split this by SIG?

Will do!

File: staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go [hunk 1]
```diff
@@ -54,19 +54,36 @@ const (
...
+}
+
 // FromString creates an IntOrString object with a string value.
 func FromString(val string) IntOrString {
 	return IntOrString{Type: String, StrVal: val}
 }
 
+// PointerFromInt32 creates an IntOrString object with an int32 value
+// and returns a pointer to it.
+func PointerFromInt32(val int32) *IntOrString {
```
[the remark this answers is not in the document]
Comment (skitt) [reply staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go:77]: Yes, perhaps we should — or rather, drop the second commit from this PR (and the others).

[the remark this answers is not in the document]
Comment (alculquicondor) [reply staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go:77]: I would vote for dropping the second commit for now, and wait for the generic pointer.

[the remark this answers is not in the document]
Comment (skitt) [reply staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go]: Done.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**